### ISSUE TITLE
Failed to spawn videos on Safari

### ISSUE THREAD

**Issue by takahirox** [mozilla/hubs#4516]
**Description**

I failed to spawn videos on Safari.

**To Reproduce**

1. Go to hubs.mozilla.com
2. Enter a video or YouTube URL in chat or "Place" menu and spawn it
3. See that not video but an audio panel shows up
4. See the console error

`Error loading video - "https://hubs-proxy.com/https://..."`

**Expected behavior**

Spawning videos work correctly

**Hardware** 
 - Device: Desktop, iPhone
 - OS: MacOS, iOS
 - Browser: Safari

**Additional context**

It works on MacOS Chrome, so it may be Safari specific problem.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/HUBS-834)

**Comment by takahirox** [comment]
Somehow I can no longer reproduce this problem on my MacOS (Big Sur 11.5.2) + Safari (14.1.2). I will close this issue so far.

**Comment by takahirox** [comment]
Reopened because @Softvision-RemusDranca seems to still face this problem.

**Comment by matthewbcool** [comment]
unable to repro on macOS- I did get some popping noise on the audio but otherwise worked as expected. 
2019 macbook safari 14.1.2

**Comment by takahirox** [comment]
**The root issue**

I speculated the problem with @keianhzo, and we found that the problem seem to be caused by the HTTP range request and wrong(?) HTTP 200 response.

Safari seems to send HTTP range request for `videoEl.src = url` and expects HTTP 206 response. If a server responds with 200, Safari causes an error.

Safari seems to have a bug that it doesn't transfer the range property in HTTP request if it sends a request with cross origin and a server redirects (302). Then the server ends up responding with 200 because it receives a regular (non-range) request. Other major browsers transfer range property.

<img width="725" alt="1" src="https://user-images.githubusercontent.com/7637832/132933162-36eba351-eda8-4e74-92d8-755845f664be.png">

<img width="723" alt="2" src="https://user-images.githubusercontent.com/7637832/132933164-5573c8e6-e810-4ccb-b7a7-b578330b96e9.png">

**Situation where the problem can cause**

In Hubs, this problem seems to be caused if (1) proxy server is `hubs-proxy.com` (lambda based one) and (2) you attempt to spawn YouTube video (because YouTube video server originally redirects).

The problem doesn't seem to be cause with reticulum proxy server like `cors-proxy-dev.reticulum.io` (JavaScript based one) because the server internally follows the redirects and doesn't respond 302 to browsers.

Hubs Cloud instance and the test environment with `npm run dev` seem to use reticulum based proxy server by default. Only hubs.mozilla.com seems to use `hubs-proxy.com`.

So the problem can happen with the combination hubs.mozilla.com + YouTube spawn + Safari.

**Workaround**

I'm going to make a simple test case reproducing the problem and send a bug report to Safari devs, but we need a workaround until the bug will be fixed.

This can be a workaround. `videoEl.srcObject` takes `Blob`.

```
const res = await fetch(url);
videoEl.srcObject = await res.blob();
```

Downside of this workaround is (if I'm right) `fetch()` loads entire file and then start to play the video so response time can be worse. Regular `videoEl.src = url` seems to send HTTP request so that (if I'm right) the chunks can be downloaded in parallel and also video can start to play before download completion. (video has `canplaythrough` event).

**Alternative**

Alternative solution may be letting the lambda based proxy server `hubs-proxy.com` internally follow redirects as the reticulum based one does, but there is a chance that it can break something. We don't really want to adopt this approach.
